includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -127,17 +145,41 @@ public function get_scripts() {
 		return $scripts;
 	}
 
+	/**
+	 * Process alternative names in attribute spec list.
+	 *
+	 * @since 0.7
+	 *
+	 * @param array $attr_spec_list Attribute spec list.
+	 * @return array Modified attribute spec list.
+	 */
+	private function process_alternate_names( $attr_spec_list ) {
```

Comment:
Actually it should just be called once for each tag. It loops over all attributes itself.

By doing it once at the constructor we avoid double-nested `foreach` loops inside the `sanitize_disallowed_attributes_in_node` method below. Since the double foreach O(n^2) was being done at the node level, it could be extremely poor for performance. So now `is_amp_allowed_attribute` is able to use the `rev_alternate_attr_name_lookup` to do fast lookups.